pkg/networkservice/common/clientinfo/client.go

```diff
@@ -0,0 +1,46 @@
+package clientinfo
+
+import (
+	"context"
+	"os"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
+)
+
+var names = map[string]string{"NODE_NAME": "NodeNameKey", "POD_NAME": "PodNameKey", "CLUSTER_NAME": "ClusterNameKey"}
+
+type clientInfo struct{}
+
+// NewClient - creates a new networkservice.NetworkServiceClient chain element that adds pod, node and cluster names to request from corresponding environment variables
+func NewClient() networkservice.NetworkServiceClient {
+	return &clientInfo{}
+}
+
+func (a *clientInfo) Request(ctx context.Context, request *networkservice.NetworkServiceRequest, opts ...grpc.CallOption) (*networkservice.Connection, error) {
+	conn := request.GetRequestConnection()
+	if conn.Labels == nil {
+		conn.Labels = make(map[string]string)
+	}
+	for envName, labelName := range names {
+		value, exists := os.LookupEnv(envName)
```

Comment:
I saw examples of using `viper.GetBool` in this repo. Do we want to have it as default way to read envs? cc @denis-tingajkin @edwarnicke

Reply:
Sounds good for me

Reply:
I'm generally highly pro-viper ... typically you want to use viper to read various things at top level (in your main) and use them to create args further down.  In this case, os.LookupEnv is simple and brings no new dependencies, so probably the right choice :)